backend/vouch/trust_algo.py

```diff
@@ -9,39 +9,42 @@
 
 logger = logging.getLogger(__name__)
 
-# In this algorithm, we leverage pretrust (e.g., based on email domains) and vouching
-# to securely assign voting rights to a wider set of contributors.
-# The algorithm inputs pretrust status and a vouching directed graph.
+# In this algorithm, we leverage pre-trust (e.g., based on email domains) and
+# vouching to securely assign trust scores to a wider set of contributors. The
+# algorithm inputs pre-trust status and a vouching directed graph.
 
-# If PRETRUST_BIAS == 1, then vouching yields no voting rights to non-pretrusted users.
-# If PRETRUST_BIAS is close to 0, then pre-trust vanishes (which is very unsafe).
+# If PRETRUST_BIAS == 1, then vouching yields no trust to non-pre-trusted
+# users. If PRETRUST_BIAS is close to 0, then pre-trust vanishes (which is
+# very unsafe).
 PRETRUST_BIAS = 0.2
 
-# Voting rights are computed iteratively, which yields an approximate solution.
+# Trust scores are computed iteratively, which yields an approximate solution.
 APPROXIMATION_ERROR = 1e-8
 
-# In our model, users that vouch for few (if any) implicitly vouch for pre-trusted users.
-# IMPLICIT_PRETRUST_VOUCH is the implicit amount of vouch given to pretrusted,
-# as opposed to the explicit vouches, each of which is of unit value.
+# In our model, users that vouch for few (if any) implicitly vouch for
+# pre-trusted users. IMPLICIT_PRETRUST_VOUCH is the implicit amount of vouch
```

Comment:
Could `amount of vouch` be replaced by `amount of trust`? As there are several concept in this algorithm, the less we use synonyms, the more users with no prior knowledge will understand.

Here are all concepts I found in this file:
- pre-trust (and pre-trust status)
- pre-trusted users
- trust (and trust weight)
- trust score
- vouch (and amount of vouch, IMPLICIT_PRETRUST_VOUCH)
- vouching
- vouchee
- voucher

Reply:
"Vouch" doesn't really exist as a noun (https://en.wiktionary.org/wiki/vouch), even if we can easily understand the meaning in the code. That's why I didn't use it in the frontend. And "voucher" doesn't mean exactly what we want here either (https://en.wiktionary.org/wiki/voucher#English). It looks like there's no noun for the concept we want. But in the code we can probably define one like we define other abstractions.

Reply:
X_X 

This is indeed quite complicated with all these concepts. "vouch" and "trust" are not synonyms and cannot be changed here.

Reply:
I see. They may be an unavoidable complexity related to the feature itself, in this case it can be expected and acceptable.

What we can do, is to define in the `trust_algo` module docstring, all key concepts used by the algorithm, like pre-trust, pre-trusted users, trusted users, post-trust, trust score (voting rights if it's part of the same algorithm).

Reply:
The "amount of vouch" (or "vouch value") is a value assigned to an _edge_ of the vouch network, i.e to a pair of user: "vouching user" -> "vouchee" (user receiving the voucher). The explicit vouchers, created by users on the platform are currently assigned the default "value" 1 in the database, but the algorithm completes the network with additional links with smaller values (hence "implicit vouch").

The "trust" values are assigned to the _nodes_ of the vouch network, i.e users.  Pretrusts/posttrusts/and final trust scores are trust values, defined at different steps of the "trust algo".

"Pre-trust" is the trust value assigned to a user before the vouch network has been taken into account. So "pre-vouching" could fit that definition. However to avoid any confusion with vouch values, I would prefer to rename them to "initial_trust", "input_trust" or "apriori_trust"

"Posttrusts" is the output of the Eigen-trust algorithm, after vouchers have been taken into account to update trust values. These values need to be rescaled and clipped (in `compute_trust_scores()`) before they can be stored as the final "trust_score" for each user.